src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -293,20 +298,23 @@ Map<String, Object> buildMapWithTargetKeyAndOriginalValue(IngestDocument ingestD
         return mapWithProcessorKeys;
     }
 
-    private void buildMapWithProcessorKeyAndOriginalValueForMapType(
+    private void buildNestedMap(
```

Comment:
Would be nice to have unit test for this method

Reply:
ack